packages/x-date-pickers/src/internals/components/PickerProvider.tsx

```diff
@@ -167,7 +167,8 @@ export interface PickerContextValue<
    * The ref to attach to the element that triggers the Picker opening.
    * When using a built-in field component, this property is automatically attached to the right element.
    */
-  triggerRef: React.RefObject<any>;
+  triggerRef: React.Dispatch<React.SetStateAction<HTMLElement | null>>;
+  triggerRefElement: HTMLElement | null;
```

Comment:
I think we can keep it on the private context.
And I would name it `triggerElement`, since it's not a ref anymore.